File: backend/db/model/Comment.go
package model

import "encoding/json"

type Comment struct {
	Base
	UserName      string
	UserID        int64
	SpreadSheetID string
	SheetNo       int64
	CellID        string
	Content       string
}

func (in *Comment) Stringify() string {
	b, err := json.Marshal(in)
	if err != nil {
		return ""
	}
	return string(b)
}

func StringifyComments(in []*Comment) string {
	b, err := json.Marshal(in)
	if err != nil {
		return ""
	}
	return string(b)
}

File: web/src/app/types/Comment.ts
export type Comment = {
    PK: string,
    SK: string,
    CreatedAt: Date,
    UpdatedAt: Date,
    DeletedAt: Date | null,
    UserName: string,
    UserID: number,
    SpreadSheetID: string,
    SheetNo: number,
    CellID: string,
    Content: string
}
